src/crate/client/connection.py

```diff
@@ -197,14 +198,20 @@ def get_blob_container(self, container_name):
 
     def _lowest_server_version(self):
         lowest = None
+        last_connection_error = None
         for server in self.client.active_servers:
             try:
                 _, _, version = self.client.server_infos(server)
                 version = Version(version)
-            except (ValueError, ConnectionError):
+            except ConnectionError as ex:
+                last_connection_error = ex
+                continue
```

Comment:
Won't this `continue` let you try all active servers? Do you want to stop at the first exception?

Reply:
Yes and no.

Reply:
I probably take the change log entry too literally.

Reply:
Sorry, so what's happening? does it retry all servers provided and then throw the last connection error? (shouldn't it throw the first connection error maybe?)

Reply:
If it would throw the first error, the whole logic of supporting multiple servers would be invalidated I think.

Reply:
ah sry got this wrong, not sure if throwing the first error would be better. both could make sense imho.

Reply:
another option could be to put all connection errors into an exception if all servers raise connection errors.

Reply:
> If it would throw the first error, the whole logic of supporting multiple servers would be invalidated I think.

I share the same opinion that it would defeat the whole purpose, right.

Reply:
@matriv: Do you like it better this way, now with 306feb5869791e?